StreetCare/StreetCare/Views/Screens/OutreachFormView.swift

```diff
@@ -0,0 +1,219 @@
+//
+//  OutreachFormView.swift
+//  StreetCare
+//
+//  Created by Marian John on 11/12/24.
+//
+import SwiftUI
+import FirebaseFirestore
+
+struct OutreachFormView: View {
+    @Environment(\.presentationMode) var presentationMode  // Environment property to dismiss view
+    @State private var title = ""
+    @State private var location = ""
+    @State private var date = Date()
+    @State private var time = Date()
+    @State private var eventDescription = ""
+    @State private var skillsDescription = ""  // Skills description field
+    @State private var showAlert = false
+    @State private var alertMessage = ""
+
+    private let db = Firestore.firestore()
+    private let titleLimit = 15
+    private let descriptionLimit = 200
+
+    var allFieldsFilled: Bool {
+        !title.isEmpty && !location.isEmpty
+    }
+
+    func saveToFirestore() {
```

Comment:
add an `isLoading` state variable here:

```
    @State private var isLoading = false
    
    func saveToFirestore() {
        guard allFieldsFilled else {
            alertMessage = .emptyFields
            showAlert = true
            return
        }

        isLoading = true
        let formData: [String: Any] = [
            "title": title,
            "location": location,
            "date": date,
            "time": time,
            "eventDescription": eventDescription,
            "skillsDescription": skillsDescription
        ]

        db.collection("outreachEventsDev").addDocument(data: formData) { error in
            isLoading = false
            if let error = error {
                alertMessage = .error
                showAlert = true
            } else {
                alertMessage = .success
                showAlert = true
            }
        }
    }
```

and at the bottom of the VStack of the view add:

```
  .overlay(
      isLoading ? ProgressView("Saving...").padding().background(Color.white.opacity(0.8)).cornerRadius(10) : nil
  )